docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+
+Troubleshooting this bug and moreover determining how to fix it in such a way that makes similar mistakes less likely in the future has lead to the following mini-framework for understanding the HFC's behavior on a particular chain.
+
+So many dimensions arose from investiating such a relatively simple misbehavior for two reasons.
+First, the mechanisms implementing these policies are---at least in the current implementation---somewhat interdependent.
+Second, the HFC (including its instantiation for Cardano) remains underdocumented since back when it was originaly developed under time-pressure, and our team no longer includes its inventor.
+The Consensus Team learns new things and unfortunately re-learns some old things every time they dive in to alter it.
+(Optimistically, Nicolas Frisby feels they currently understand it now better than they have ever before.)
+
+## Dimensions
+
+The following discrete variables characterize dimensions of the HFC design space.
+In this context, the _determination_ is the `singleEraTransition` method: it's how an era's ledger state determines when the era ends.
+
+- DST - Determination is Sensitive to Ticks  - whether ticking the ledger state can change its determination of the end of its era: Sensitive vs Insensitive
+
+- CCT - Criss-Crossed Ticks - whether an era's tick rules might be used to tick across slots that are in _other_ eras: Allowed vs Disallowed
+
+- PIL - Potential to Ignore the Ledger - whether the HFC can ignore the ledger's hard fork-related governance: Possible vs Impossible
+
+- DSR - Determination is Sensitive to Rollback - for chains C and D where C < D and len(C) <= len(C /\ D) + k, whether the current era of C can have a determined end on C but an undetermined or different end on D: Sensitive vs Insensitive
+
+- ZEE - Zero Epoch Eras - whether an era (in its entirety) can contain zero epochs: Allowed vs Disallowed
+
+- ZBE - Zero Block Eras - whether an era (in its entirety) can contain zero blocks: Allowed vs Disallowed
+
+- QSR - Queries are Sensitivity to Rollback - same as DSR but for HFC-related Node-To-Client queries instead of for validity of blocks: Sensitive vs Insensitive
+
+Note the following relations.
+
+- ZEE=Allowed implies ZBE=Allowed.
+- DSR=Insensitive implies QSR=Insensitive.
+
+## Table for the Current Cardano HFC
+
+The following table and associated notes classifies the current HFC implementation in terms of these dimensions.
+
+| DST | CCT | PIL | DSR | ZEE | ZBE | QSR |
+| --- | --- | --- | --- | --- | --- | --- |
+| I   | A   | P 1 | I   | 2   | 3   | I   |
+
+- (1) It's Possible, but only if there's a Chain Growth violation at a particular time during the epoch.
```

Comment:
What does (1) refer to?

Reply:
I would need an example to understand this :grimacing:

Reply:
It refers to the `1` in the PIL column right above (maybe it could also use (1), or the GH markdown footnote syntax?).

> I would need an example to understand this 😬

See the description in https://github.com/input-output-hk/ouroboros-consensus/pull/346#discussion_r1354586938 (less than $k$ blocks in the last $6k/f$ slots of an epoch)

Reply:
Thanks for the link to the comment. It's clear now (provided I understood what said comment tried to explain :)).